pootle/apps/pootle_project/models.py

```diff
@@ -510,6 +524,24 @@ def _get_code(self, project):
     ### /TreeItem
 
 
+@receiver([vfolder_post_save, pre_delete])
+def invalidate_resources_cache_for_vfolders(sender, instance, **kwargs):
+    if instance.__class__.__name__ == 'VirtualFolder':
+        try:
+            # In case this is vfolder_post_save.
+            affected_projects = kwargs['projects']
+        except KeyError:
+            # In case this is pre_delete.
+            affected_projects = Project.objects.filter(
+                translationproject__stores__unit__vfolders=instance
+            ).distinct().values_list('code', flat=True)
+
+        cache.delete_many([
+            make_method_key('Project', 'resources', proj)
+            for proj in affected_projects
```

Comment:
> and calculating the resources is not expensive.

Correcting my own words: this indeed now has become super expensive compared to the previous way resources were retrieved (half a second vs a few ms)